apps/prairielearn/src/question-servers/freeform.ts

```diff
@@ -1409,20 +1544,31 @@ export async function render(
 
       // Gather dependencies for all rendered elements
       allRenderedElementNames.forEach((elementName) => {
-        let resolvedElement = resolveElement(elementName, context);
-        const elementDependencies = structuredClone(resolvedElement.dependencies) ?? {};
-        const elementDynamicDependencies =
-          structuredClone(resolvedElement.dynamicDependencies) ?? {};
+        const resolvedElement = resolveElement(elementName, context);
+
+        const elementDependencies: (ElementCourseDependencyJson | ElementCoreDependencyJson) & {
```

Comment:
This is weird to me, I am doing this wrong I think.

Reply:
I reworked this in 9f2dcbc. In general, TypeScript is happiest when one isn't shoving random properties into objects, which is what we were doing before.

Reply:
This seems much cleaner.